Assets/_BForBoss/_Utility/Editor/PropertyDrawers/ResolverPropertyDrawer.cs

```diff
@@ -0,0 +1,369 @@
+                        }
+                    }
+                    else
+                    {
+                        components = mb.GetComponentsInParent(fieldType, _includeInactiveGameObjects)
+                            .Where(component => component.gameObject != mb.gameObject).ToArray();
+                    }
+                    
+                    break;
+                }
+                case ResolveType.FromSelf:
+                {
+                    if (isGameObjectType)
+                    {
+                        gameObjects = new List<GameObject>{mb.gameObject};
+                    }
+                    else
+                    {
+                        components = mb.GetComponents(fieldType);
+                    }
+                    
+                    break;
+                }
+            }
+            
+            void OnItemSelected(int componentID)
+            {
+                property.serializedObject.Update();
+                property.objectReferenceInstanceIDValue = componentID;
+                property.serializedObject.ApplyModifiedProperties();
+            }
+
+            if (isGameObjectType)
+            {
+                if (gameObjects.IsNullOrEmpty())
+                {
+                    return;
+                }
+
+                if (gameObjects.Count > 1)
+                {
+                    var dropdown = new GameObjectResolverDropdown(gameObjects, OnItemSelected, new AdvancedDropdownState());
+                    dropdown.Show(_childResolveRect);
+                }
+                else
+                {
+                    property.serializedObject.Update();
+                    property.objectReferenceValue = gameObjects[0];
+                    property.serializedObject.ApplyModifiedProperties();
+                }
+            }
+            else
+            {
+                if (components.IsNullOrEmpty())
+                {
+                    return;
+                }
+
+                if (components.Length > 1)
+                {
+                    var dropdown = new ComponentResolverDropdown(components, OnItemSelected , new AdvancedDropdownState());
+                    dropdown.Show(_childResolveRect);
+                }
+                else
+                {
+                    property.serializedObject.Update();
+                    property.objectReferenceValue = components[0];
+                    property.serializedObject.ApplyModifiedProperties();
+                }
+            }
+        }
+
+        private void InitializeContent()
+        {
+            _childResolveContent = new GUIContent("C", "Get Component from Children");
+            _parentResolveContent = new GUIContent("P", "Get Component from Parent");
+            _selfResolveContent = new GUIContent("S", "Get Component from Self");
+            _clearContent = EditorGUIUtility.IconContent("d_winbtn_mac_close_h@2x", "|Clear Component");
```

Comment:
>"|Clear Component"
Do you need the "|"

Reply:
Yeah. To be honest I have no idea why haha!

https://docs.unity3d.com/ScriptReference/EditorGUIUtility.IconContent.html

Reply:
>character to flag it as a tooltip.
what the fuck, why not just build it in.